elements/pl-dropdown/pl-dropdown.py

```diff
@@ -0,0 +1,174 @@
+import prairielearn as pl
+import lxml.html
+import random
+import chevron
+from enum import Enum
+
+WEIGHT_DEFAULT = 1
+BLANK_ANSWER = ' '
+
+
+class SortTypes(Enum):
+    RANDOM = 'random'
+    ASCEND = 'ascend'
+    DESCEND = 'descend'
+
+
+def get_options(element, data, correct_answer):
+    options = []
+    for child in element:
+        if child.tag in ['pl-answer']:
+            pl.check_attribs(child, required_attribs=[], optional_attribs=['correct'])
+            child_html = pl.inner_html(child).strip()
+            options.append(child_html)
+    return options
+
+
+def get_solution(element, data, correct_answer):
+    solution = []
+    for child in element:
+        if child.tag in ['pl-answer']:
+            pl.check_attribs(child, required_attribs=[], optional_attribs=['correct'])
+            is_correct = pl.get_boolean_attrib(child, 'correct', False)
+            child_html = pl.inner_html(child).strip()
+            if is_correct:
+                solution = child_html
+    return solution
+
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, required_attribs=['correct-answer'], optional_attribs=['blank', 'weight', 'sort'])
+    correct_answer = pl.get_string_attrib(element, 'correct-answer')
+
+    # Get answer from pl-answer if implemented
+    data['correct_answers'][correct_answer] = get_solution(element, data, correct_answer)
+
+    # Get answer from server.py if pl-answer not implemented
+    if correct_answer is None:
+        data['correct_answers'][correct_answer] = correct_answer
+
+    if data['correct_answers'][correct_answer] is None:
+        raise Exception('Correct answer not defined for correct-answer: %s' % correct_answer)
+
+
+def render(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    correct_answer = pl.get_string_attrib(element, 'correct-answer')
+    dropdown_options = get_options(element, data, correct_answer)
+
+    sort_type = pl.get_string_attrib(element, 'sort', '').upper().strip()
+    blank_start = pl.get_boolean_attrib(element, 'blank', False)
```

Comment:
Need to add at the top of the file:

```python
BLANK_DEFAULT = False
```

Then use:

```suggestion
    blank_start = pl.get_boolean_attrib(element, 'blank', BLANK_DEFAULT)
```

Reply:
Fixed.